jetty-http/src/main/java/org/eclipse/jetty/http/HttpParser.java

```diff
@@ -1034,14 +1031,14 @@ else if (_endOfContent == EndOfContent.CHUNKED_CONTENT)
                             _field = new HostPortHttpField(_header,
                                 _compliances.contains(HttpComplianceSection.FIELD_NAME_CASE_INSENSITIVE) ? _header.asString() : _headerString,
                                 _valueString);
-                            addToConnectionTrie = _fieldCache != null;
+                            addToFieldCache = true;
                         }
                         break;
 
                     case CONNECTION:
                         // Don't cache headers if not persistent
-                        if (HttpHeaderValue.CLOSE.is(_valueString) || new QuotedCSV(_valueString).getValues().stream().anyMatch(HttpHeaderValue.CLOSE::is))
-                            _fieldCache = null;
+                        if (_handler.getHeaderCacheSize() > 0 && (HttpHeaderValue.CLOSE.is(_valueString) || _valueString.contains(HttpHeaderValue.CLOSE.asString())))
```

Comment:
`String.contains()` is case sensitive, so this test would fail for `_valueString="Close,Other"`, right?
I think using the previous version was expensive, but it was on the last expression of the `||` so executed only rarely.

Reply:
Let me convert to the HttpField.contains method, which is not expensive